packages/nx/src/command-line/release/utils/git.ts

```diff
@@ -558,3 +564,16 @@ export async function getFirstGitCommit() {
     throw new Error(`Unable to find first commit in git history`);
   }
 }
+
+export async function getGitRoot() {
+  try {
+    return (await execCommand('git', ['rev-parse', '--show-toplevel'])).trim();
+  } catch (e) {
+    throw new Error('Unable to find git root');
+  }
+}
+
+async function getWorkspaceRelativePath() {
+  const gitRoot = await getGitRoot();
+  return relative(gitRoot, workspaceRoot);
+}
```

Comment:
The git root is as constant as the workspaceRoot for the lifecycle of the program, let's avoid recomputing this (and running the child process) every time the getGitDiff function is called.

```suggestion
let gitRootRelativePath: string;
async function getGitRootRelativePath() {
  if (!gitRootRelativePath) {
    const gitRoot = await getGitRoot();
    gitRootRelativePath = relative(gitRoot, workspaceRoot);
  }
  return gitRootRelativePath;
}
```